databricks/koalas/series.py

```diff
@@ -323,6 +323,9 @@ def __init__(self, data=None, index=None, dtype=None, name=None, copy=False, fas
                 assert not fastpath
                 s = data
             else:
+                if name is None:
+                    self._has_no_name = True
+                self._has_no_name = False
```

Comment:
I don't think we should have this attribute. e.g.,

```py
>>> kser = ks.Series([1,2,3])
>>> kser.name = 'foo'
>>> kser
???  # maybe always None
```

How about just setting name for `self` after `IndexOpsMixin.__init__(...)`?

Reply:
ahh, definitely better!!